src/main/java/htsjdk/samtools/util/Histogram.java

```diff
@@ -42,40 +35,46 @@
  *
  * @author Tim Fennell
  */
-public class Histogram<K extends Comparable> extends TreeMap<K, Bin> {
+public final class Histogram<K extends Comparable<K>> implements Serializable {
```

Comment:
I don't know if you want to enforce that K is also serializable.  It seems like a reasonable thing to add to me.